repository/SmalltalkCI-Pharo-Core.package/SmalltalkCIPharo9.class/instance/preLoad.st

```diff
@@ -0,0 +1,15 @@
+executing
+preLoad
+	[ (IceRepositoryCreator new
+		location: projectDirectory asFileReference;
+		createRepository) register. ] 
+		on: Error 
+		do: [ :e |
+		      	 self class printWarningToStdout: (String streamContents: [ :s |
+			 	s << 'Could not register the tested repository with Iceberg.';
```

Comment:
Is this always printed, including when someone does not need Iceberg and the repo was only shallow cloned? If so, maybe we should silence the error here? Any thoughts, @theseion?

Reply:
You're right. It's not bad but a bit ugly. I wonder whether we could make registration an option of the run script? E.g.,:
```bash
./run.sh -s Pharo64-10 --headful -d ../repo/.smalltalk.ston --register
```

I guess, the GitHub action would then have the equivalent option.

Reply:
Good point. The config file is probably a better place for the setting.

Reply:
That's what I'd tend to do.
However, I'd have to investigate how to do it.
Does that sounds okay @theseion ?

Reply:
For the implementation, maybe take a look at https://github.com/hpi-swa/smalltalkCI/commit/0a4eb665b36f7e46c866b4e03c189330f33b384a.

Reply:
Sounds good. I would expand the comment a bit and either point to documentation on what "full history" means or explain the necessary steps (it's clear what it means for a local repository but not for a CI system).

Reply:
> For the implementation, maybe take a look at [0a4eb66](https://github.com/hpi-swa/smalltalkCI/commit/0a4eb665b36f7e46c866b4e03c189330f33b384a).

Thanks for the pointer, I will try to do it after my phd defense.